packages/xrpl/test/client/autofill.test.ts

```diff
@@ -32,17 +34,112 @@ describe('client.autofill', function () {
       TransactionType: 'DepositPreauth',
       Account: 'rGWrZyQqhTp9Xu7G5Pkayo7bXjH4k4QYpf',
       Authorize: 'rpZc4mVfWUif9CRoHRKKcmhu1nx2xktxBo',
+      NetworkID,
       Fee,
       Sequence,
       LastLedgerSequence,
     }
     const txResult = await testContext.client.autofill(tx)
 
+    assert.strictEqual(txResult.NetworkID, NetworkID)
     assert.strictEqual(txResult.Fee, Fee)
     assert.strictEqual(txResult.Sequence, Sequence)
     assert.strictEqual(txResult.LastLedgerSequence, LastLedgerSequence)
   })
 
+  it('ignores network ID if missing', async function () {
+    const tx: Payment = {
+      TransactionType: 'Payment',
+      Account: 'XVLhHMPHU98es4dbozjVtdWzVrDjtV18pX8yuPT7y4xaEHi',
+      Amount: '1234',
+      Destination: 'X7AcgcsBL6XDcUb289X4mJ8djcdyKaB5hJDWMArnXr61cqZ',
+      Fee,
+      Sequence,
+      LastLedgerSequence,
+    }
+    testContext.mockRippled!.addResponse('ledger', rippled.ledger.normal)
+
+    const txResult = await testContext.client.autofill(tx)
+
+    assert.strictEqual(txResult.NetworkID, undefined)
+  })
+
+  it('overrides network ID if > 1024 and version is later than 1.11.0', async function () {
+    testContext.client.networkID = 1025
+    testContext.client.buildVersion = '1.11.1'
+    const tx: Payment = {
+      TransactionType: 'Payment',
+      Account: 'XVLhHMPHU98es4dbozjVtdWzVrDjtV18pX8yuPT7y4xaEHi',
+      Amount: '1234',
+      Destination: 'X7AcgcsBL6XDcUb289X4mJ8djcdyKaB5hJDWMArnXr61cqZ',
+      Fee,
+      Sequence,
+      LastLedgerSequence,
+    }
+    testContext.mockRippled!.addResponse('ledger', rippled.ledger.normal)
+
+    const txResult = await testContext.client.autofill(tx)
+
+    assert.strictEqual(txResult.NetworkID, 1025)
+  })
+
+  it('ignores network ID if > 1024 but version is earlier than 1.11.0', async function () {
+    testContext.client.networkID = 1025
+    testContext.client.buildVersion = '1.10.1'
```

Comment:
Is it possible to write this test to use a different `server_info` mock instead of changing the values by hand?